meshtastic/ota.py

```
@@ -97,44 +127,62 @@ def update(
+            file_hash = hashlib.sha256()
+            for block in iter(
+                lambda: firmware.read(FILE_HASH_READ_CHUNK_SIZE_BYTES), b""
+            ):
+                file_hash.update(block)
+            self._assert_session_firmware_unchanged(
+                current_size=size, current_hash=file_hash
+            )
+            firmware.seek(0)
+
+            if OTA_PROGRESS_LOG_PERCENT_STEP <= 0:
+                raise ValueError("OTA_PROGRESS_LOG_PERCENT_STEP must be > 0")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)